### PR TITLE
TRUNK-5182: Refactored code within saveConcept and created new privat…

### PR DESCRIPTION
**Pull request by jrs33**
…e methods listed below saveConcept in order of their calls. Did not remove code from saveConcept yet to avoid any bugs that would come along with an incorrect refactor.

https://issues.openmrs.org/browse/TRUNK-5182

### CODE REVIEW

**Comment by dkayiwa** [comment]
@jrs33 did you take a look at the review comments from codacy-bot? 😊

**Comment by jrs33** [comment]
@dkayiwa Yes! So I think those flags may be coming from the fact that I did not actually refactor the code to ease the pull request review as mentioned on the ticket at 

https://issues.openmrs.org/browse/TRUNK-5182

I simply marked where my newly added private methods would replace code in saveConcept, so I think thats where the unused private methods warning comes from. Do you notice anything other than that that I may have missed?

**Comment by dkayiwa** [comment]
@jrs33 what is the value of adding new private methods that you never use?

**Comment by jrs33** [comment]
"And only extract the private methods without refactoring them as well in order to ease code review and reduce risk of breaking the code. If you have an idea on how to improve the private methods later on submit another PR." This was the exact reasoning I saw on the open ticket. I can resubmit with the methods in place in saveConcept. Would you prefer that?

**Comment by dkayiwa** [comment]
Extracting private methods without using them anywhere?

**Comment by jrs33** [comment]
@dkayiwa just added in the private methods instead of excluding them as was mentioned on the ticker

**Comment by dkayiwa** [comment]
@jrs33 can you squash the commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by jrs33** [comment]
@dkayiwa squashed the two commits into one. Please let me know of any other issues and Ill be happy to change them!

File: api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java [hunk 1]
```diff
@@ -207,6 +197,23 @@ private void ensureConceptMapTypeIsSet(Concept concept) {
 		}
 	}
 
+	private void voidNameMakeSynonym(ConceptName conceptName) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: The method name seems to suggest that you are also voiding the name. Can you change it to match the comments?

File: api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java [hunk 2]
```diff
@@ -207,6 +197,23 @@ private void ensureConceptMapTypeIsSet(Concept concept) {
...
+		// Make the voided name a synonym, this would help to avoid
+		// having multiple fully specified or preferred
+		// names in a locale in case the name is unvoided
+		if (!conceptName.isSynonym()) {
+			conceptName.setConceptNameType(null);
+		}
+	}
+
+	private void falsifyLocalePreferred(ConceptName conceptName) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: falsify does look so much tied to the method implementation details. I would go for something like makeLocaleNotPreferred or something else along those lines.

File: api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java [hunk 3]
```diff
@@ -207,6 +197,23 @@ private void ensureConceptMapTypeIsSet(Concept concept) {
...
+		// Sets preferred locale of the Concept to false unless
+		// the Concept is already false
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java]: I do not see the value of the above comment.

**Comment by dkayiwa** [comment]
@jrs33 are you still working on this?

**Comment by jrs33** [comment]
@dkayiwa Sorry about the delay, was busy finishing up my internship project this past week along with a website, so got a bit sidetracked. Working on adding those finishing touches now!

**Comment by dkayiwa** [comment]
@jrs33 thanks for the update! Looking forward to the finishing touches. 😊

**Comment by jrs33** [comment]
@dkayiwa Think the changes should be all set now. Is the request all set to be pulled in?

**Comment by dkayiwa** [comment]
@jrs33 the ticket status says you are still doing some more work on this. 😊

**Comment by jrs33** [comment]
@dkayiwa I initialized another code review, so please let me know if there are any other changes that need to be done. Thanks!

**Comment by dkayiwa** [comment]
@jrs33 can you squash the commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by jrs33** [comment]
@dkayiwa I think it should be set. I did a rebase and have the following in my log: 
---
Joshuas-MacBook-Air:openmrs-core joshuasurette$ git log
commit 2a9ae6ee18361e5b254326e1e1aabb86cb2db306
Author: jrs33 <surettej@bu.edu>
Date:   Thu Aug 3 10:18:09 2017 -0400

    TRUNK-5182: Refactored code within saveConcept and created new private methods listed below saveConcept in order of their calls. Did not remove code from saveConcept yet to avoid any bugs that would come along with an incorrect refactor.
    
    refactoring code and implementing private methods within saveConcept
    
    Fixing incorrectly unit test name that was causing errors and adding more descriptive function names
---
Seems to have squashed properly. Let me know if anything else needs to be done

**Pull request merged**